### DISCUSSION TITLE
Can I build for x86_64 on an arm64 device?

### DISCUSSION THREAD

**Discussion by TechStudent10** [joshiemoore/snakeware#166] [Q&A] [answered]
Just a question, can I build for x86_64 on an arm64 device?

_Originally posted by @TechStudent11 in https://github.com/joshiemoore/snakeware/issues/162#issuecomment-939354882_

**Comment by MikeDev101** [comment]
Sadly, no (at least, logically). x86-64 and arm64 are two radically different architectures. It's roughly the same as asking a fish to climb a tree, and asking a squirrel to go diving.

**Comment by joshiemoore** [comment, marked as answer]
Cross-compilation should work fine, that's a big part of what buildroot does. Just as we can build images targeting the RPi on x86, you should be able to build for x86 from an ARM host.

I'm closing this because I just enabled the "Discussions" tab so people can ask questions there instead of opening issues.

Edit: I migrated it to the "Discussions" tab, didn't know you could do that. This is the first time I've enabled Discussions on one of my repos :^)

**Reply by TechStudent10** [reply]
@joshiemoore Well, I tried compiling for x86_64 on my Raspberry Pi 4 machine, but in over 12 hours I didn't get an ISO. The terminal just hanged. And I re-cloned it so I don't know.

**Reply by TechStudent10** [reply]
> I would think that attempting a full snakeware build on an RPi host could take like a day or more, I wouldn't recommend using an RPi as your dev environment for this.
> 
> https://elinux.org/Raspberry_Pi_Kernel_Compilation
> 
> The "Overview" section of this page says that compiling a kernel on an RPi took them 12.5 hours, and snakeware builds a lot more than just the Linux kernel.

Legit just saw this. Might try doing it in a VM.
